src/main/scala/scala/tools/refactoring/analysis/SymbolExpanders.scala

```diff
@@ -205,4 +207,37 @@ trait DependentSymbolExpanders extends TracingImpl {
       case _ => Nil
     })
   }
+
+  /**
+   * Associates term symbols with missing ranges to related symbols that have ranges.
+   *
+   * The reason that we need this is that in some cases, the PC generates multiple
+   * symbols for one and the same symbol in user source code, one of them with a
+   * proper range position, and others just with offset positions. One place where
+   * this happens is in desugared for comprehensions with filter clauses.
+   * See Assembler Ticket #1002650.
+   */
+  trait TermsWithMissingRanges extends SymbolExpander { this: IndexLookup =>
+    protected abstract override def doExpand(s: Symbol): List[Symbol] = {
+      termsWithSamePointButRange(s) ::: super.doExpand(s)
+    }
+
+    private def termsWithSamePointButRange(s: Symbol): List[Symbol] = {
+      val lookAtSymbol = {
+        s != NoSymbol && !s.isSynthetic && s.isInstanceOf[TermSymbol] &&
+          s.pos.isInstanceOf[OffsetPosition] && !s.pos.isInstanceOf[RangePosition]
```

Comment:
just asking, maybe 2.10 api is too weak so you decided to use
`isInstanceOf`?

2016-04-08 9:46 GMT+02:00 Matthias Langer <notifications@github.com>:

> In src/main/scala/scala/tools/refactoring/analysis/SymbolExpanders.scala
> <https://github.com/scala-ide/scala-refactoring/pull/156#discussion_r58990634>
> :
>
> > +   *
> > +   * The reason that we need this is that in some cases, the PC generates multiple
> > +   * symbols for one and the same symbol in user source code, one of them with a
> > +   * proper range position, and others just with offset positions. One place where
> > +   * this happens is in desugared for comprehensions with filter clauses.
> > +   * See Assembler Ticket #1002650.
> > +   */
> > +  trait TermsWithMissingRanges extends SymbolExpander { this: IndexLookup =>
> > +    protected abstract override def doExpand(s: Symbol): List[Symbol] = {
> > +      termsWithSamePointButRange(s) ::: super.doExpand(s)
> > +    }
> > +
> > +    private def termsWithSamePointButRange(s: Symbol): List[Symbol] = {
> > +      val lookAtSymbol = {
> > +        s != NoSymbol && !s.isSynthetic && s.isInstanceOf[TermSymbol] &&
> > +          s.pos.isInstanceOf[OffsetPosition] && !s.pos.isInstanceOf[RangePosition]
>
> Thanks for pointing that out! See #157
> <https://github.com/scala-ide/scala-refactoring/pull/157>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/scala-ide/scala-refactoring/pull/156/files/44270070343f8811cc3e9524e9fe35921b3b1f1c#r58990634>
>